src/main/kotlin/no/nav/familie/oppdrag/simulering/mock/Periode.kt

```diff
@@ -0,0 +1,84 @@
+package no.nav.familie.oppdrag.simulering.mock
+
+import java.math.BigDecimal
+import java.time.DayOfWeek
+import java.time.LocalDate
+
+class Periode : Comparable<Periode> {
+    var fom: LocalDate
+        private set
+    var tom: LocalDate
+    var sats: BigDecimal? = null
+        private set
+    var oldSats: BigDecimal? = null
+        private set
+    var periodeType: PeriodeType? = null
+    var kodeKlassifik: String? = null
+        private set
+
+    internal constructor(fom: LocalDate, tom: LocalDate) {
+        this.fom = fom
+        this.tom = tom
+    }
+
+    internal constructor(fom: LocalDate, tom: LocalDate, sats: BigDecimal?, kodeKlassifik: String?) {
+        this.fom = fom
+        this.tom = tom
+        this.sats = sats
+        this.kodeKlassifik = kodeKlassifik
+    }
```

Comment:
Disse 2 trengs ikke hvis man setter default null på sats, kodeKlassifik, periodeType på de andre constructors.
For å unngå private set så er det mulig å bruke 

```kotlin
data class Periode private constructor(val fom: LocalDate,
                                       var tom: LocalDate,
                                       val sats: BigDecimal?,
                                       val oldSats: BigDecimal?,
                                       var periodeType: PeriodeType?,
                                       val kodeKlassifik: String?) : Comparable<Periode> {

    constructor(fom: LocalDate,
                tom: LocalDate,
                sats: BigDecimal? = null,
                kodeKlassifik: String? = null,
                periodeType: PeriodeType? = null) :
            this(fom = fom,
                 tom = tom,
                 sats = sats,
                 oldSats = null,
                 periodeType = periodeType,
                 kodeKlassifik = kodeKlassifik)

    constructor(fom: LocalDate,
                tom: LocalDate,
                oldSats: BigDecimal,
                sats: BigDecimal?,
                kodeKlassifik: String?) :
            this(fom = fom,
                 tom = tom,
                 sats = sats,
                 oldSats = null,
                 periodeType = if (oldSats.compareTo(sats) <= 0) {
                     PeriodeType.ØKNING
                 } else {
                     PeriodeType.REDUKSJON
                 },
                 kodeKlassifik = kodeKlassifik)
```

Reply:
Endret ihht review comment 😄